python/taichi/lang/impl.py

```diff
@@ -199,10 +199,25 @@ def subscript(value, *_indices, skip_reordered=False, get_ref=False):
                     f"Gradient {_var.get_expr_name()} has not been placed, check whether `needs_grad=True`"
                 )
         field_dim = snode.num_active_indices()
+
         if field_dim != index_dim:
```

Comment:
Under `field_dim == index_dim` it is still possible that `field_dim == 1` and a 1-element vector is passed in. We need to handle the special case before we compare the dimensions.